### PR TITLE
D3ASIM-3677: Fixed time_stamp handling for orders in markets.

### PR DESCRIPTION
**Pull request by hannesdiedrich**

### CODE REVIEW

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by fievelk — DISMISSED** [review]
LGTM

File: tests/market/test_market_redis_connection.py [hunk 1]
```diff
@@ -214,7 +216,7 @@ def test_subscribes_to_market_channels(self):
         )
 
     def test_accept_bid_calls_market_method_and_publishes_response(self):
-        bid = Bid("b_id", now(), 12, 13, "b_buyer")
+        bid = Bid("b_id", now, 12, 13, "b_buyer")
```
Comment (Hassan754) [inline tests/market/test_market_redis_connection.py:219]: Why did you remove the calling parentheses ?

File: tests/market/test_market.py [hunk 2]
```diff
@@ -117,37 +117,34 @@ def test_market_offer_delete_readonly(market):
...
-    assert trade.creation_time == now
+    assert trade.creation_time > e_offer.creation_time
     assert trade.offer_bid == e_offer
     assert trade.seller == "A"
     assert trade.buyer == "B"
 
 
 def test_balancing_market_negative_offer_trade(market=BalancingMarket(
-    bc=NonBlockchainInterface(str(uuid4())), time_slot=now())):  # NOQA
+        bc=NonBlockchainInterface(str(uuid4())))):  # NOQA
```
Comment (Hassan754) [inline tests/market/test_market.py:141]: Why is there an extra indentation here?

File: integration_tests/steps/storage_test.py [hunk 3]
```diff
@@ -42,30 +42,44 @@ def check_storage_prices(context):
...
-                    for trade in trades_sold])
-        assert all([round(trade.offer_bid.energy_rate, 2) <=
-                    round(final_buying_rate[trade.creation_time.hour], 2)
-                    for trade in trades_bought])
         assert len(trades_sold) > 0
+        assert len(trades_bought) > 0
+        for trade in trades_sold:
+            expected_time_stamp = start_time_stamp.add(hours=trade.creation_time.hour)
+            assert (round(trade.offer_bid.energy_rate, 2) >=
+                    round(final_selling_rate[expected_time_stamp], 2))
```
Comment (Hassan754) [inline integration_tests/steps/storage_test.py:77]: Nit, IMO we can put this code directly inside the `for trade in market.trades:` loop, no need to loop 3 times

**Review by Hassan754 — APPROVED** [review]
LGTM

File: tests/market/test_market.py (hunk 2 above)
Comment (spyrostz) [reply tests/market/test_market.py:141]: Because it is on the same indentation level as the function body and some versions of Pycharm complain AFAIR.

File: integration_tests/steps/storage_test.py (hunk 3 above)
Comment (spyrostz) [reply integration_tests/steps/storage_test.py:77]: Fixed

**Review by fievelk — APPROVED** [review]

File: tests/test_redis_communication.py [hunk 4]
```diff
@@ -253,11 +253,12 @@ def test_event_listener_calls_the_event_listener_of_the_root_dispatcher(self, ma
...
-        offer = Offer("1", now(), 2, 3, "A")
-        trade = Trade("2", now(), Offer("accepted", now(), 7, 8, "Z"), "B", "C",
+        now = datetime(2021, 11, 3, 10, 45)
+        offer = Offer("1", now, 2, 3, "A")
+        trade = Trade("2", now.add(minutes=1), Offer("accepted", now, 7, 8, "Z"), "B", "C",
                       None, None, TradeBidOfferInfo(None, None, None, None, None))
-        new_offer = Offer("3", now(), 4, 5, "D")
-        existing_offer = Offer("4", now(), 5, 6, "E")
+        new_offer = Offer("3", now, 4, 5, "D")
+        existing_offer = Offer("4", now, 5, 6, "E")
```
Comment (Hassan754) [inline tests/test_redis_communication.py:261]: Why did we remove the now function calls?

**Review by Hassan754 — COMMENTED** [review]
one comment, LGTM

**Review by faizan2590 — APPROVED** [review]

**Review by Hassan754 — APPROVED** [review]

File: tests/test_redis_communication.py (hunk 4 above)
Comment (spyrostz) [reply tests/test_redis_communication.py:261]: In favor of `now = datetime(2021, 11, 3, 10, 45)` in line 256
